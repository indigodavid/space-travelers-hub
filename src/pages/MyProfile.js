import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const missions = useSelector((state) => state.missions);
  const rockets = useSelector((state) => state.rockets);
  // filter missions and rockets accordingly
  return (
    <div className="profile">
      <div className="profile-column">
        <h2>My Missions</h2>
        <ul id="missions-list">
          {missions.map((mission) => (
            <li key={mission.mission_id}>{mission.mission_name}</li>
          ))}
        </ul>
      </div>
      <div className="profile-column">
        <h2>My Rockets</h2>
        <ul id="rockets-list">
          {rockets.map((rocket) => (
            <li key={rocket.id}>{rocket.rocket_name}</li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
