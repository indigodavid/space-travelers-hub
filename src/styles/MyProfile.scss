/* stylelint-disable at-rule-no-unknown */
@mixin profile-lists {
  display: flex;
  flex-direction: column;
  gap: 1em;
  align-items: center;
  text-align: center;
}

#missions-list {
  @include profile-lists();
}

#rockets-list {
  @include profile-lists();
}

.profile {
  margin: 2em auto;
  display: grid;
  grid-template-columns: 1fr 1fr;

  h2 {
    text-align: center;
    margin-bottom: 1em;
  }
}
