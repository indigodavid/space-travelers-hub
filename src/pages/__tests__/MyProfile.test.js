import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore'
import MyProfile from '../MyProfile';

const renderMyProfilePage = () => {
  const { container } = render(
    <Provider store={store}>
      <MyProfile />
    </Provider>
  );
  return container;
}

describe('Test MissionsPage page', () => {
  test('Match Snapshot', () => {
    expect(renderMyProfilePage().firstChild).toMatchSnapshot();
  })

  test('Find Elements ny text', () => {
    renderMyProfilePage();
    expect(screen.getByText(/My Rockets/i)).toBeInTheDocument();
    expect(screen.getByText(/My Missions/i)).toBeInTheDocument();
  })
})

