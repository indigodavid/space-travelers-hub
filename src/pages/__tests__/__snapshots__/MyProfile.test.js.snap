// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MissionsPage page Match Snapshot 1`] = `
<section
  class="tableContainer"
>
  <div
    class="missionsProfile"
  >
    <h1>
      My Missions
    </h1>
    <div
      class="itemsList"
    >
      <li>
        There are no missions joined.
      </li>
    </div>
  </div>
  <div
    class="rockets"
  >
    <h1>
      My Rockets
    </h1>
    <div
      class="itemsList"
    >
      <li>
        There are no rockets reserved.
      </li>
    </div>
  </div>
</section>
`;
